### ISSUE TITLE
How can i reproduce the results in kitti

### ISSUE THREAD

**Issue by cvsod** [macandro96/pytorch-depth-motion#1]
Thanks for your work!
I retrain the model in kitti 26epochs, however, i sample image from tensorbard 
![1](https://github.com/macandro96/pytorch-depth-motion/assets/93752061/c6c7ca20-b881-4b2b-a952-45bd9f846989)
It seems that residual translation learns nothing, and the pred-depth maps have holes,
I retrain the model use 8GPUS, The config as follow

`
    name: depth_motion_bs8_lr2e-4  # name of the experiment
    
    mode: training
    pretrained_name: null 
    
    
    grid:
      index: -1  
    dataset_name: KITTI
    
    model:
      motion_burnin_steps: 20000  # number of steps to freeze the motion model
  
  
    depth_model:
      pretrained: True
      use_layernorm: False
      num_layers: 18
      layernorm_noise_rampup_steps: 10000
  
  
    motion_model:
      input_shape: [128, 416]
      learn_intrinsics: True  # currently only supports True
  
  
    loss_weights:
      depth_supervision: 0.0
      depth_variance: 0.0
      depth_smoothing: 0.001
      rgb_consistency: 1.0
      depth_consistency: 0.0
      ssim: 3.0
      rotation_cycle_consistency: 1e-3
      translation_cycle_consistency: 5e-2
      motion_smoothing: 1.0
      motion_sparsity: 0.2
  
    optimizer:
      name: adam # optimizer to use for training. Can be ['adamw', 'adam']
      lr: 2e-4
      betas: [0.9, 0.999]
      weight_decay: 0
  
      scheduler:
        name: StepLR  # Can be ['CosineAnnealing', 'Linear', 'StepLR', 'ReduceLROnPlateau']
  
  
        warmup_steps: null
        warmup_ratio: 0.1
        min_lr: 1e-5
  
        # StepLR parameters
        step_size: 10
        gamma: 0.1
  
        # ReduceLROnPlateau
        factor: 0.8
        patience: 5
  
    dataset:
      batch_size: 8
      num_workers: 10
      shuffle: true
      evaluateKITTIFlow: False  # if True, will also evaluate and log EPE on KITTI Flow dataset.
      kitti_flow_path: null  # path to the kitti dataset, will be used if the above param is true.
    
      kitti:
        path: "/horizon-bucket/matrix2/users/meijuan.jia/mono_workspace/kitti_data" 
        img_ext: ".png"  # can be .png, .jpg
        num_scales: 1
        side: null
        dedup_threshold: 0  # condition to dedup -> |rgb_{t} - rgb{t+1}| < dedup_threshold
        stride: 1  # stride while sampling images
    
    trainer:
      accelerator: auto
      devices: auto
      num_nodes: 1
      max_epochs: 30
      check_val_every_n_epoch: 1  
      strategy: ddp_find_unused_parameters_false
      accumulate_grad_batches: 1
      benchmark: true
      enable_progress_bar: true
      log_every_n_steps: 50  # Interval of logging
      enable_checkpointing: true
      limit_train_batches: 1.0
      limit_val_batches: 1.0
      sync_batchnorm: true
      num_sanity_val_steps: 0
      enable_model_summary: false 
      gradient_clip_val: 10
  
  
    exp_manager:
      exp_dir: ./log  # path to the directory where the logs and checkpoints will be saved
      exp_name: ${name}
    
      logging:
        mask_flow: false
        log_image_interval_train: 1000  # number of steps after which image will be logged in train mode
        log_image_interval_val: 500  # number of steps after which image will be logged when in val mode
    
      checkpoint_callback_params:
        monitor: val_loss
        save_top_k: 1
        save_last: true
    
      create_wandb_logger: false
      wandb_logger_kwargs:
        name: ${exp_manager.exp_name}
        version: null
        project: null
        offline: false  # if true, wandb logging will be done offline and would require manual syncing
        tags: null  # list of tags to assign to the run
    
    hydra:
      run:
        dir: ${exp_manager.exp_dir}/${exp_manager.exp_name}
      sweep:
        dir: ${hydra.run.dir}
        subdir: ${hydra.job.num}`
`
How can i reproduce the result?

**Comment by macandro96** [comment]
Thanks for reaching out. Can you try increasing the number of epochs? I had to train for ~ 100 epochs to get the depth maps as shown in the repo. 
Also - try switching off the learning rate scheduler. I remember that our depth maps weren't as good when we used StepLR.

**Comment by macandro96** [comment]
Closing this issue for now as it has been sometime. Feel free to reopen if the problem persists.
